src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -784,17 +807,16 @@ private async Task EnsureInitializeAsync()
                     {
                         await OnSolutionExistsAndFullyLoadedAsync();
                     }
+
+                    _initialized = true;
                 }
-                else
+                catch (Exception)
                 {
-                    // Check if the cache is initialized.
-                    // It is possible that the cache is not initialized, since,
-                    // the solution was not saved and/or there were no projects in the solution
-                    if (!_cacheInitialized && _solutionOpenedRaised)
-                    {
-                        await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
-                        await EnsureNuGetAndVsProjectAdapterCacheAsync();
-                    }
+                    throw;
```

Comment:
I often do this for debugging, but this does nothing except make exception handling slower at runtime (unless the compiler has been improved to optimize it away). Delete it from the production code.

Reply:
But if I don't re-throw here then it swallows exception and we'll never see the exception at higher level.
Another reason I'm using try catch is make sure SemaphoreLock still released if something happen.